cmd/daemon/player.go

```diff
@@ -560,13 +546,72 @@ func (p *AppPlayer) handleApiRequest(ctx context.Context, req ApiRequest) (any,
 	}
 }
 
+func (p *AppPlayer) handleMprisEvent(ctx context.Context, req mpris.MediaPlayer2PlayerCommand) error {
+	switch req.Type {
+	case mpris.MediaPlayer2PlayerCommandTypeNext:
+		return p.skipNext(ctx, p.state.tracks.PeekNext(ctx))
+	case mpris.MediaPlayer2PlayerCommandTypePrevious:
+		return p.skipPrev(ctx, true)
+	case mpris.MediaPlayer2PlayerCommandTypePlay:
+		return p.player.Play()
+	case mpris.MediaPlayer2PlayerCommandTypePause:
+		return p.player.Pause()
+	case mpris.MediaPlayer2PlayerCommandTypePlayPause:
+		if p.state.player.IsPaused {
+			return p.player.Play()
+		} else {
+			return p.player.Pause()
+		}
+	case mpris.MediaPlayer2PlayerCommandTypeStop:
+		return p.Stop(ctx)
+	case mpris.MediaPlayer2PlayerCommandLoopStatusChanged:
+		log.Tracef("mpris loop status argument %s", req.Argument)
+		dt := req.Argument
+		t := true
+		f := false
```

Comment:
This is a bit ugly, but I see why you are doing it. I wonder if a `ptr` helper would make it better.